.github/scripts/check-react-native-log.sh

```diff
@@ -0,0 +1,94 @@
+#!/bin/bash
+
+echo "cd build-system-tests/mega-apps/${MEGA_APP_NAME}"
+cd build-system-tests/mega-apps/${MEGA_APP_NAME}
+
+# Define the time of seconds to wait the command
+if [ "$PLATFORM" = "ios" ]; then
+  TIME_TO_WAIT=300
+else
+  TIME_TO_WAIT=200
+fi
+
+# Define the color codes
+BLUE_BOLD="\033[1;36m"
+GREEN_BOLD="\033[1;32m"
+RED_BOLD="\033[1;31m"
+YELLOW_BOLD="\033[1;33m"
+NC="\033[0m"
+
+# Sleep for the specified time
+echo -e "${BLUE_BOLD}Sleep for $TIME_TO_WAIT second...${NC}"
+sleep $TIME_TO_WAIT
+
+# Check if the log file only has one line
+if [[ $(wc -l <"$LOG_FILE") -eq 1 && $(head -n 1 "$LOG_FILE") == "info Starting logkitty" ]]; then
+  echo -e "${RED_BOLD}Failed to get the logging messages. Please increase TIME_TO_WAIT.${NC}"
+  exit 1
+fi
+
+# Define the exceptions to the error criteria
+EXCEPTIONS=("AuthError -")
+START_MESSAGES=("info Starting logkitty" "React Native iOS Logger started for XCode project")
+
+# Step 0: Check if the log file has only one line with the start message
+echo -e "${BLUE_BOLD}Checking log file for start messages...\n${NC}"
+if [[ $(wc -l <"$LOG_FILE") -eq 1 ]]; then
+  for start_message in "${START_MESSAGES[@]}"; do
+    if echo $(head -n 1 "$LOG_FILE") | grep -q "$start_message"; then
+      echo -e "${RED_BOLD}Failed to get the logging messages. Please increase TIME_TO_WAIT.${NC}"
+      echo -e "${BLUE_BOLD}Full log:${NC}"
+      cat "$LOG_FILE"
+      exit 1
+    fi
+  done
+fi
+
+# Step 1: Check if all lines in the log file meet the criteria:
+echo -e "${BLUE_BOLD}Checking log file \"$LOG_FILE\" for errors...${NC}"
+LINE_ERROR=false
+
+while read -r line; do
+  if [[ "$line" == "" ]]; then
+    continue
+  # Check if there's any line start with "NN:NN:NN, " (N is 0-9), and has "Error" or "ERROR".
+  elif echo "$line" | grep -Eq '^([0-9]{2}:){2}[0-9]{2}, .*(Error|ERROR)'; then
+    echo -e "${RED_BOLD}ERROR found:${NC}"
```

Comment:
`[ERROR]` or `Error:` are from React Native, so we can't control the format, just have to detect them.

The colors are for us to easily locate the bash commands in Github Workflow.

I'm not sure where "info" and "[INFO]" are, but those are not generated by us either.

Reply:
Oh, I was talking about the logs we log with `echo -e`. For example, in line 56 above we use both color code and "ERROR" prefix, whereas in other code we don't add string prefix and just use color code.

I did misread the code though, and see that `info` string prefix that I was talking about was external, thanks for clarifying that..

So what I was proposing that we prefix what we log with severity in square brackets (`[INFO]`, `[ERROR]`) along with color coding so that it is both easily locatable and text searchable. What do you think?

Reply:
Good idea! Done 91e5207af